src/libraries/System.IO.Compression/src/System/IO/Compression/ZipHelper.cs

```diff
@@ -17,20 +18,22 @@ internal static class ZipHelper
 
         private static readonly DateTime s_invalidDateIndicator = new DateTime(ValidZipDate_YearMin, 1, 1, 0, 0, 0);
 
-        internal static bool RequiresUnicode(string test)
+        internal static Encoding GetEncoding(string text)
         {
-            Debug.Assert(test != null);
-
-            foreach (char c in test)
+            if (!string.IsNullOrEmpty(text))
```

Comment:
Can it be null? The assert was asserting it cannot. And it's not `string?`
If not then checking for empty may not be interesting either as it will just skip the loop.

Reply:
You're right, my mistake. The two places where this `GetEncoding` method is called, I ensure the passed text is not null. Plus, it's not nullable, so I would've been warned anyway.

Neither the `string IsNullorEmpty` nor the `Debug.Assert` checks are necessary. I'll remove them.